ocaml/xcp-rrdd/bin/rrdp-netdev/rrdp_netdev.ml

```diff
@@ -132,18 +133,16 @@ let transform_taps devs =
     newdevnames
 
 let generate_netdev_dss () =
-  let _, doms, _ =
-    Xenctrl.with_intf (fun xc -> Xenctrl_lib.domain_snapshot xc)
-  in
-
-  let uuid_of_domid domains domid =
-    let _, uuid, _ =
-      try List.find (fun (_, _, domid') -> domid = domid') domains
-      with Not_found ->
-        failwith
-          (Printf.sprintf "Failed to find uuid corresponding to domid: %d" domid)
-    in
-    uuid
+  let uuid_of_domid domid =
+    try
+      Xenstore.with_xs (fun xs ->
+          let vm = xs.Xenstore.Xs.getdomainpath domid ^ "/vm" in
+          let vm_dir = xs.Xenstore.Xs.read vm in
+          xs.Xenstore.Xs.read (vm_dir ^ "/uuid")
+      )
+    with _ ->
+      failwith
```

Comment:
```
let fail = Printf.ksprintf failwith (* somewhere at the top of the file *)

..
fail "Something %s" "a string"
```

Might want to log the exception to understand why this fails. I would use `Filename.concat` to construct the paths but it's a aesthetic choice.